File: templates/Pages/Settings.CodeBehind/Services/ThemeSelectorService.cs
using System;
using Windows.Storage;
using Windows.UI.Xaml;
using Param_RootNamespace.Helpers;
using System.Threading.Tasks;

namespace Param_RootNamespace.Services
{
    public static class ThemeSelectorService
    {
        private const string SettingsKey = "RequestedTheme";

        public static event EventHandler<ElementTheme> OnThemeChanged = delegate { };

        public static bool IsLightThemeEnabled => Theme == ElementTheme.Light;
        public static bool IsDarkThemeEnabled => Theme == ElementTheme.Dark;
        
        public static ElementTheme Theme { get; set; }

        public static async Task InitializeAsync()
        {
            Theme = await LoadThemeFromSettingsAsync();
        }

        public static async Task SetThemeAsync(int themeId)
        {
            ElementTheme theme;
            switch (themeId)
            {
                case 0:
                    theme = ElementTheme.Light;
                    break;
                case 1:
                    theme = ElementTheme.Dark;
                    break;
                case 2:
                default:
                    theme = ElementTheme.Default;
                    break;
            }

            await SetThemeAsync(theme);
        }

        public static int GetTheme()
        {
            switch (Theme)
            {
                case ElementTheme.Light:
                    return 0;
                case ElementTheme.Dark:
                    return 1;
                case ElementTheme.Default:
                default:
                    return 2;
            }
        }

        public static async Task SetThemeAsync(ElementTheme theme)
        {
            Theme = theme;
            SetRequestedTheme();
            await SaveThemeInSettingsAsync(Theme);
            OnThemeChanged(null, Theme);
        }

        public static void SetRequestedTheme()
        {
            var frameworkElement = Window.Current.Content as FrameworkElement;
            if (frameworkElement != null)
            {
                frameworkElement.RequestedTheme = Theme;
            }
        }

        private static async Task<ElementTheme> LoadThemeFromSettingsAsync()
        {
            ElementTheme cacheTheme = ElementTheme.Default;
            string themeName = await ApplicationData.Current.LocalSettings.ReadAsync<string>(SettingsKey);
            if (!String.IsNullOrEmpty(themeName))
            {
                Enum.TryParse<ElementTheme>(themeName, out cacheTheme);
            }
            return cacheTheme;
        }

        private static async Task SaveThemeInSettingsAsync(ElementTheme theme)
        {
            await ApplicationData.Current.LocalSettings.SaveAsync<string>(SettingsKey, theme.ToString());
        }
    }
}

File: templates/Pages/SettingsVB/Services/ThemeSelectorService.vb
Imports Windows.Storage

Imports Param_RootNamespace.Helpers

Namespace Services
    Public NotInheritable Class ThemeSelectorService
        Private Sub New()
        End Sub

        Private Const SettingsKey As String = "AppBackgroundRequestedTheme"
        
        Public Shared Property Theme As ElementTheme = ElementTheme.Default

        Public Shared Async Function InitializeAsync() As Task
            Theme = Await LoadThemeFromSettingsAsync()
        End Function

        Public Shared Async Function SetThemeAsync(theme As ElementTheme) As Task
            ThemeSelectorService.Theme = theme

            SetRequestedTheme()
            Await SaveThemeInSettingsAsync(Theme)
        End Function

        Public Shared Sub SetRequestedTheme()
            If TypeOf Window.Current.Content Is FrameworkElement Then
                Dim frameworkElement = TryCast(Window.Current.Content, FrameworkElement)
                frameworkElement.RequestedTheme = Theme
            End If
        End Sub

        Private Shared Async Function LoadThemeFromSettingsAsync() As Task(Of ElementTheme)
            Dim cacheTheme As ElementTheme = ElementTheme.[Default]
            Dim themeName As String = Await ApplicationData.Current.LocalSettings.ReadAsync(Of String)(SettingsKey)

            If Not String.IsNullOrEmpty(themeName) Then
                [Enum].TryParse(themeName, cacheTheme)
            End If

            Return cacheTheme
        End Function

        Private Shared Async Function SaveThemeInSettingsAsync(theme As ElementTheme) As Task
            Await ApplicationData.Current.LocalSettings.SaveAsync(SettingsKey, theme.ToString())
        End Function
    End Class
End Namespace
